services/src/main/java/org/keycloak/forms/account/freemarker/model/RealmBean.java

```diff
@@ -72,4 +76,9 @@ public boolean isRegistrationEmailAsUsername() {
     public boolean isUserManagedAccessAllowed() {
         return realm.isUserManagedAccessAllowed();
     }
+
+    public boolean isUserDeleteAccountAllowed() {
+       RequiredActionProviderModel deleteAction = realm.getRequiredActionProviderByAlias("delete_account");
+        return Objects.nonNull(realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).getRole(AccountRoles.DELETE_ACCOUNT)) && Objects.nonNull(deleteAction) && deleteAction.isEnabled();
+    }
```

Comment:
Change the logic here to checking if action is enabled, and if user has the delete account role